server/scopeserver/api/v1/users.py

```diff
@@ -1,17 +1,86 @@
 " API endpoints related to managing users. "
 
-from fastapi import APIRouter, Depends
+from typing import List
+
+from fastapi import APIRouter, Depends, HTTPException, Response, status
 from sqlalchemy.orm import Session
 
-from scopeserver import crud, schemas
+from scopeserver import crud, models, schemas, oidc
 from scopeserver.api import deps
 
 router = APIRouter()
 
-# pylint: disable=invalid-name
+
+@router.post("/new", summary="Create a new guest user.", response_model=schemas.Token)
+async def new_user(database: Session = Depends(deps.get_db)):
+    """Create a new guest user."""
+    new_guest = schemas.UserResponse.from_orm(crud.create_user(database=database))
+    access_token = oidc.create_access_token(data=new_guest)
+    return schemas.Token(  # nosemgrep: gitlab.bandit.B106
+        access_token=access_token,
+        token_type="bearer",
+        user=new_guest,
+    )
+
+
+@router.get("/", summary="Get all accessible projects", response_model=List[schemas.Project])
+async def my_projects(
+    database: Session = Depends(deps.get_db),
+    current_user: models.User = Depends(deps.get_current_user),
+):
+    """Retrieve all projects for the current user."""
+    return crud.get_projects_for_user(database=database, user_id=current_user.id)
+
+
+@router.get("/all", summary="Get all users", response_model=List[schemas.UserResponse])
+async def all_users(
+    database: Session = Depends(deps.get_db),
+    offset: int = 0,
+    limit: int = 100,
+    _current_user: models.User = Depends(deps.get_current_admin),
+):
+    "Get a listing of all users"
+    return crud.get_users(database, offset, limit)
+
+
+@router.post("/promote", summary="Promote a user to an admin", response_model=schemas.UserResponse)
+async def promote(
```

Comment:
Should this not require the submitting user to _be_ an admin to promote to admin? Or is this checked elsewhere?